MediaPickerController.swift

```diff
@@ -0,0 +1,183 @@
+//
+//  MediaPickerController.swift
+//  MediaPickerController
+//
+//  Created by Pablo Villar on 3/9/16.
+//  Copyright © 2016 Inaka. All rights reserved.
+//
+
+import UIKit
+import AVFoundation
+import MobileCoreServices
+
+enum MediaPickerControllerType {
+	case ImageOnly
+	case ImageAndVideo
+}
+
+@objc protocol MediaPickerControllerDelegate {
+	optional func mediaPickerControllerDidPickImage(image: UIImage)
+	optional func mediaPickerControllerDidPickVideoWithURL(url: NSURL, imageData: NSData, thumbnail: UIImage)
+}
+
+class MediaPickerController: NSObject {
+	
+	// MARK: - Public
+	
+	weak var delegate: MediaPickerControllerDelegate?
+	
+	init (type: MediaPickerControllerType, presentingViewController controller: UIViewController) {
+		self.type = type
+		self.presentingController = controller
+		self.mediaPicker = UIImagePickerController()
+		super.init()
+		self.mediaPicker.delegate = self
+	}
+	
+	func show() {
+		let actionSheet = self.optionsActionSheet
+		self.presentingController.presentViewController(actionSheet, animated: true, completion: nil)
+	}
+	
+	// MARK: - Private
+	
+	private let presentingController: UIViewController
+	private let type: MediaPickerControllerType
+	private let mediaPicker: UIImagePickerController
+	
+}
+
+extension MediaPickerController: UIImagePickerControllerDelegate, UINavigationControllerDelegate {
+	
+	// MARK: - UIImagePickerControllerDelegate
+	
+	func imagePickerController(picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : AnyObject]) {
+		self.dismiss()
+		let mediaType = info[UIImagePickerControllerMediaType] as! NSString
+		
+		if mediaType.isEqualToString(kUTTypeImage as NSString as String) {
```

Comment:
Checked. It's needed because `kUTTypeImage` is defined as `CFString`

`public let kUTTypeImage: CFString`

Good find though.